common/Environments/Scripts/HyperVSetupScript.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DevHome.Common.Environments.Scripts;
+
+public static class HyperVSetupScript
+{
+    public const string SetupFunction = @"
```

Comment:
Can this be put into a `.ps1` file and `SetupFunction` read from that file?

Reply:
So I decided to do it as a string because we run this elevated. I'm afraid of some bad acting coming along and editing the ps1 file and use blinding running it. At least this way the string is set at compile time

Reply:
If a bad actor could write into DevHome's install location we'd be in really bad shape. What would prevent from replacing some binaries? Or DevSetupAgent.zip that we install on VMs? Or the whole DevHome?

Only System and TrustedInstaller have full access to packaged app install locations. No one else has Write permissions. Even an elevated admin cannot write anything without taking ownership of the location and changing permissions first. But to do that you'd need your code running elevated and there would be no need to change some script when you can do whatever you want already.

My concern is we bring PS for tasks that can be done with native APIs. Error surface increases, diagnosability decreases, performance decreases. I understand we don't have much time for these fixes, but if that's the only reason to use PS we should refactor it in the future.